src/ragapp/backend/middlewares/rate_limit.py

```diff
@@ -2,29 +2,49 @@
 import time
 
 import jwt
-from backend.services.user_chat_service import UserChatService
 from fastapi import HTTPException, Request, status
 from fastapi.responses import Response
 from jwt import InvalidTokenError
+from pydantic import BaseModel, computed_field
+
+from backend.services.user_chat_service import UserChatService
 
 JWT_COOKIE_NAME = "Authorization"  # The name of the cookie that stores the JWT token
 JWT_USER_ID_CLAIM = "preferred_username"  # The claim in the JWT token that stores the user ID or user name
-CHAT_REQUEST_LIMIT_THRESHOLD = int(os.environ.get("CHAT_REQUEST_LIMIT_THRESHOLD", 5))
+JWT_USER_ROLES_CLAIM = (
+    "X-Forwarded-Roles"  # The claim in the JWT token that stores the user roles
+)
+CHAT_REQUEST_LIMIT_THRESHOLD = int(os.getenv("CHAT_REQUEST_LIMIT_THRESHOLD", 0))
+CHAT_REQUEST_LIMIT_ENABLED = CHAT_REQUEST_LIMIT_THRESHOLD > 0
+ADMIN_ROLE = "admin-manager"
+
+
+class UserInfo(BaseModel):
+    username: str
+    roles: list[str] = []
+
+    @computed_field
+    def is_admin(self) -> bool:
+        return ADMIN_ROLE in self.roles
 
 
 async def request_limit_middleware(request: Request) -> Response:
-    time_frame = _get_time_frame()
-    user_id = _extract_user_id_from_request(request)
-    request_count = UserChatService.get_user_chat_request_count(user_id, time_frame)
-    if request_count >= CHAT_REQUEST_LIMIT_THRESHOLD:
-        raise HTTPException(
-            status_code=status.HTTP_429_TOO_MANY_REQUESTS,
-            detail=f"You have exceeded {CHAT_REQUEST_LIMIT_THRESHOLD} chat requests. Please try again later.",
+    if CHAT_REQUEST_LIMIT_ENABLED:
+        time_frame = _get_time_frame()
+        user = _extract_user_info_from_request(request)
```

Comment:
good idea with `UserInfo` - but it's not a concern of the rate limiting, please move it to a own file (it's a model) and use it here like:

```suggestion
        user = UserInfo.extract_from_request(request)
```

also move the `extract_from_request` function below then

Reply:
the code will be way cleaner as a lot of constants will also end up in the new file

Reply:
Good idea! But i think this is better:
```python
jwt = JWT(request.cookies)
user = UserInfo.from_jwt_data(jwt.data)
```
The thing is we may end up with other jwt data in middleware so it's good to keep the jwt here and use it to construct user info